### ISSUE TITLE
Cant get UsePaging to work when using Configure(IObjectTypeDescriptor descriptor) with EntityFramework

### ISSUE THREAD

**Issue by ebjornset** [ChilliCream/hotchocolate#4703]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I'm using HotChocolate v 12.6.0 (.AspNetCore, .Data and .Data.EntityFramework) with EF Core v 5.0.0 (.EntityFrameworkCore and .EntityFrameworkCore.SqlServer). 

I would like to use paging, and based o the docs I've tried to add .UsePaging() in my Query subclass of ObjectType's Configure(IObjectTypeDescriptor descriptor => descriptor.Field(...). fluent code chain.

No matter where in the fluent IObjectFieldDescriptor code chain I put .UsePaging() I end up with an exception saying: 
"Unable to infer the element type from the current resolver. This often happens if the resolver is not an iterable type like IEnumerable, IQueryable, IList etc. Ensure that you either explicitly specify the element type or that the return type of your resolver is an iterable type."

I must admit that I don't know if this is a bug or a limitation, since the paging example for EF Core uses an annotation based approach and I'm using code first.

### Steps to reproduce

```
public class Query : ObjectType
  {
      protected override void Configure(IObjectTypeDescriptor descriptor)
      {
          descriptor.Field("somethings").UsePaging()
              .UseDbContext<SomenDbContext>().Resolve(ctx => ctx.DbContext<SomeDbContext>().Somethings)
              .UseProjection().UseFiltering().UseSorting();
      }
  }

Statup:
  services.AddPooledDbContextFactory<SomeDbContext>(options => options.UseSqlServer(_dBConnectionString));
  services
      .AddGraphQLServer()
      .AddQueryType<Query>()
      .AddProjections()
      .AddFiltering()
      .AddSorting()
      ;

```

### Product

Hot Chocolate

### Version

12.6.0

**Comment by tobias-tengler** [comment]
I think this because you are not specifying the GraphQL type. Try specifying the GraphQL type of whatever an item of `SomeThings` is to `UsePaging`, e.g. `.UsePaging<SomeThingType>()`.

I'm closing this for now, but feel free to re-open if this does not address your issue. :)

**Comment by ebjornset** [comment]
Thanks for the quick answer. 👍 

Using `.UsePaging<SomeThingType>()` wouldn't compile: 
"The type 'SomeThingType' cannot be used as type parameter 'TNodeType' in the generic type or method 'PagingObjectFieldDescriptorExtensions.UsePaging<TNodeType>(IObjectFieldDescriptor, Type?, GetCursorPagingProvider?, NameString?, PagingOptions)'. There is no implicit reference conversion from 'SomeThingType' to 'HotChocolate.Types.IOutputType'

But using `.UsePaging(typeof(SomeThingType))` did the trick, so my issue is solved 🥇 

I had to move UsePaging after the UseDbContext, but the error message I explained that in a very clear way. Here's the whole fluent code chain I ended up with:
```
descriptor.Field("somethings")
              .UseDbContext<SomenDbContext>().Resolve(ctx => ctx.DbContext<SomeDbContext>().Somethings)
              .UsePaging().UseProjection().UseFiltering().UseSorting();
```
